eventlet/patcher.py

```diff
@@ -320,6 +323,18 @@ def is_monkey_patched(module):
         getattr(module, '__name__', None) in already_patched
 
 
+def _green_existing_locks(threading):
+    if sys.version[0] != '2':
+        return
+    import eventlet.green.threading
+    lock_type = type(threading.Lock())
+    rlock_type = type(threading.RLock())
+    gc = __import__('gc')
+    for o in gc.get_objects():
+        if isinstance(o, rlock_type) and isinstance(o._RLock__block, lock_type):
+            o._RLock__block = eventlet.green.threading.Lock()
```

Comment:
Not sure, but this seems like a disaster if somebody holds a lock when `monkey_patch()` is called.